--- idempiere-server.sh	2024-08-30 18:16:02.920598240 +0400
+++ idempiere-server.sh	2024-08-30 17:30:43.786724260 +0400
@@ -13,8 +13,10 @@
   echo Set JAVA_HOME to the directory of your local JDK.
 fi
 
+DEBUG_PORT=${DEBUG_PORT:-4554}
+
 if [ "$1" = "debug" ]; then
-  DEBUG="-Xdebug -Xnoagent -Xrunjdwp:transport=dt_socket,address=4554,server=y,suspend=n"
+  DEBUG="-Xdebug -Xnoagent -Xrunjdwp:transport=dt_socket,address=$DEBUG_PORT,server=y,suspend=n"
 fi
 
 echo ===================================
@@ -23,7 +25,7 @@
 
 # if don't set from service get default value
 TELNET_PORT=${TELNET_PORT:-12612}
-HOST=${HOST:-localhost}
+TELNET_HOST=${TELNET_HOST:-0.0.0.0}
 
 
 VMOPTS="-Dorg.osgi.framework.bootdelegation=sun.security.ssl
@@ -32,7 +34,7 @@
 -Djetty.base=$BASE/jettyhome
 -Djetty.etc.config.urls=etc/jetty.xml,etc/jetty-deployer.xml,etc/jetty-ssl.xml,etc/jetty-ssl-context.xml,etc/jetty-http.xml,etc/jetty-https.xml,etc/jetty-threadpool.xml,etc/jetty-http-forwarded.xml
 -Dorg.apache.cxf.osgi.http.transport.disable=true
--Dosgi.console=$HOST:$TELNET_PORT
+-Dosgi.console=$TELNET_HOST:$TELNET_PORT
 -Dmail.mime.encodefilename=true
 -Dmail.mime.decodefilename=true
 -Dmail.mime.encodeparameters=true
@@ -52,4 +54,8 @@
 --add-exports java.sql.rowset/com.sun.rowset=ALL-UNNAMED
 --add-exports java.naming/com.sun.jndi.ldap=ALL-UNNAMED"
 
-"$JAVA" ${DEBUG} $IDEMPIERE_JAVA_OPTIONS $VMOPTS -jar "$BASE"/plugins/org.eclipse.equinox.launcher_1.*.jar -application org.adempiere.server.application
+IDEMPIERE_JAVA_OPTIONS=${JAVA_OPTS:-$IDEMPIERE_JAVA_OPTIONS}
+
+echo "Staring iDempiere: $JAVA ${DEBUG} $IDEMPIERE_JAVA_OPTIONS $VMOPTS -jar $BASE/plugins/org.eclipse.equinox.launcher_1.*.jar -application org.adempiere.server.application"
+
+$JAVA ${DEBUG} $IDEMPIERE_JAVA_OPTIONS $VMOPTS -jar $BASE/plugins/org.eclipse.equinox.launcher_1.*.jar -application org.adempiere.server.application
